sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs

```diff
@@ -0,0 +1,174 @@
+                    node != parentConcatenation.Right)
+                {
+                    return;
+                }
+
+                var firstStringInConcatenation = GetFirstStringFromConcatenation(parentConcatenation);
+                var beforeCurrentString = GetPenultimateString(parentConcatenation);
+                if (!firstStringInConcatenation.IsKind(SyntaxKind.StringLiteralExpression) ||
+                    !StartsWithSqlKeyword(((LiteralExpressionSyntax)firstStringInConcatenation).Token.ValueText.Trim()) ||
+                    !beforeCurrentString.IsKind(SyntaxKind.StringLiteralExpression))
+                {
+                    return;
+                }
+
+                var previousStringLastChar = ((LiteralExpressionSyntax)beforeCurrentString).Token.ValueText.ToCharArray().Last();
+                var currentStringFirstChar = node.Token.ValueText[0];
+                if (IsKnownCharacter(previousStringLastChar) && IsKnownCharacter(currentStringFirstChar))
+                {
+                    var nodeFirstToken = node.Token.ValueText.Trim().Split(' ').First();
+                    this.context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation(), nodeFirstToken));
+                }
+            }
+
+            // Given a concatenation "a" + "b" + "c", this will return the "b" syntax node (one before last)
+            private static SyntaxNode GetPenultimateString(BinaryExpressionSyntax concatenation)
+            {
+                var penultimateElement = concatenation.Left;
+                while (penultimateElement is BinaryExpressionSyntax previousAddition &&
+                    previousAddition.IsKind(SyntaxKind.AddExpression))
+                {
+                    penultimateElement = previousAddition.Right;
+                }
+                return penultimateElement;
+            }
+
+            // Given a concatenation "a" + "b" + "c", this will return the "a" syntax node
+            private static SyntaxNode GetFirstStringFromConcatenation(BinaryExpressionSyntax concatenation)
+            {
+                var firstElement = concatenation.Left;
+                while (firstElement is BinaryExpressionSyntax toTheLeftAddition &&
+                    firstElement.IsKind(SyntaxKind.AddExpression))
+                {
+                    firstElement = toTheLeftAddition.Left;
+                }
+                return firstElement;
+            }
+
+            private static bool StartsWithSqlKeyword(string firstString) =>
+                SqlKeywords.Any(s => firstString.StartsWith(s, StringComparison.OrdinalIgnoreCase));
+
+            /**
+             * The '@' symbol is used for named parameters.
+             * The '*' is a widely used wildcard.
+             * We ignore other non-alphanumeric characters (e.g. '>','=') to avoid false positives.
+             */
+            private static bool IsKnownCharacter(char c) => char.IsLetterOrDigit(c) || c == '@' || c == '*';
```

Comment:
I'm not 100% agree with `*` because some database accepts `SELECT*FROM table1` (tested on Oracle 11g).
But on this one, let's see if users complain.